src/app/shared/scroll-with-loading/scroll-with-loading.directive.ts

```diff
@@ -0,0 +1,46 @@
+import {Directive, EventEmitter, HostListener, Output} from '@angular/core';
+
+export enum LoadDirection {
+    Top,
+    Bottom,
+}
+
+@Directive({
+    selector: '[appScrollWithLoading]',
+})
+export class ScrollWithLoadingDirective {
+    @Output() loadData = new EventEmitter<LoadDirection>();
+
+    private lastScrollTop = true;
+    private lastScrollBottom = false;
+
+    @HostListener('scroll', ['$event'])
+    onScroll(event: Event) {
+        const target = event.target as HTMLInputElement;
+        const scrollTop = target.scrollTop;
```

Comment:
Тут я бы рекомендовал чуть иначе поработать с аргументами, чтобы была более строгая типизация и отсутствовали приведения типов) Плюс удобней было бы получить значения из `target`.
```suggestion
    @HostListener('scroll', ['$event.target'])
    onScroll({scrollTop, scrollHeight, clientHeight}: HTMLElement) {
```

Reply:
Исправила :blush: